mdaviz/mainwindow.py

```diff
@@ -19,36 +21,41 @@ def __init__(self):
         self.setup()
 
     def setup(self):
-        self._folderPath = None
-        self._folderName = None
-        ###
-        self._folderList = None
-        self._mdaFileName = None
-        ###
-        self._folderSize = None
-        self._mdaFileList = None
+        self._dataPath = None       # the combined data path obj (folder.parent + subfolder)
+        self._folderPath = None     # the path obj from pull down 1
+        self._folderList = None     # the list of folder in pull down 1
+        self._subFolderPath = None  # the subfolder path obj selected in pull down 2
+        self._subFolderList = None  # the list of subfolder in pull down 2
+        self._mdaFileList = None    # the list of mda file NAME str (name only)
+        self._mdaFileLen = None     # the number of mda files in the list
         self.mvc_folder = None
+        self.checkMdaFiles = True
     
         self.setWindowTitle(APP_TITLE)
         self.setRecent(None)
         self.actionOpen.triggered.connect(self.doOpen)
         self.actionAbout.triggered.connect(self.doAboutDialog)
         self.actionExit.triggered.connect(self.doClose)
-        # TODO: set up open dialog for folder
+        try:
+            self.open.released.disconnect()
```

Comment:
What situations require this code?

Reply:
I'll explain tomorrow.

Reply:
https://doc.qt.io/qtforpython-5/PySide2/QtCore/Signal.html#PySide2.QtCore.Signal.disconnect